src/script/Deployment/TaskWorker/run.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
```

Comment:
Indeed `run.sh` is a `start.sh` w/o arguments. Main reason is to allow that `./start.sh` prints help info instead of running the image. Yes, the main thing that I do not like of current TW/Publisher start is that if I forget to add my option it starts to run. Now it does that also if I mistype, that may be fixable. Also looking at crabserver, I'd rather not touch the existing /data/run.sh, and I was looking for some uniform behavior

Reply:
Well, my worry is that there is a lot of code shared between those two files. e.g. checking if symlink exits and maintaining list of symlinks. 

> Yes, the main thing that I do not like of current TW/Publisher start is that if I forget to add my option it starts to run. Now it does that also if I mistype, that may be fixable.

what if simply make it work like this:
```
./start.sh -s  ===> just run it (s stands for start);
./start.sh -d  ===> run in debug mode (d stands for debug);
./start.sh -r  ===> run from GH repo (r stands for repo);
./start.sh -k  ===> unimplemented option/typo, don't start and print help;
./start.sh ===> nothing is provided - don't start and print help;
```
of course we can change letters from `s d r` to smth else or use long options instead of short.

Reply:
I agree with the spirit. Maybe as simple as
```
LapSB:tmp$ cat run.sh 
#!/bin/bash
sh start.sh --run
LapSB:tmp$ 
```
?
I prefer longer options, but can stick with one-letter one. In case maybe better -r for "run now" and -G for "run from GitHub", I think it is easy to make this case independent (-g or -G are the same). And if no option, print help.

Reply:
But why do we still need to have `run.sh`? is it because


> Also looking at crabserver, I'd rather not touch the existing /data/run.sh, and I was looking for some uniform behavior

?
  we can update this line https://github.com/dmwm/CRABServer/blob/master/Docker/Dockerfile#L29 to start TW/Publisher using `-r` option. Or do you foresee in the future some other use cases for `run.sh`?

Reply:
crabserver's /data/run.sh does a lot of K8s related things which should stay there and that must not be done again every time we start/stop services inside the contrainer.
I suspect that it is some kind of soft standard that a container has a run.sh command to execute when it is started.
But all in all.. everything I put inside `if [ -v SERVICE] ... fi` should be in `start.sh`. Should be in `run.sh`, w/o the if, and only be executed when the container starts. When you are inside the container debuggin/developing/hacking.. you want a simple start/stop w/o messing with links.
Maybe that's the cleanest way ?

Reply:
I had forgotten a crucial "not". Have now edited previous comment.
@ddaina let me know what you think. I start to like the idea of moving creating links away from `start.sh`

Reply:
I think idea to have run.sh script which is executed only once at a container start time makes sense. I was actually thinking about it some time ago. But was not sure if the use case that I had at that time (which I don't remember now) was strong enough. And now I wanted to make sure that I understand what you are proposing. Now I get it, thanks.

First of all, I agree that symlink part should be moved from `start.sh` to `run.sh`. Further, in my opinion, we should create only one `run.sh` script - make it uniform for TW and Publisher. Then use $SERVICE variable that comes from Docker to fill in missing parts.  And maybe here we could use what is described in https://github.com/dmwm/CRABServer/issues/6206. I could have a look how to achieve it. 

If to push it even more, maybe we can have one `start.sh` script, but that is next step I guess.

Reply:
Maybe `jinja2` is an overkill for this. I like the idea of a single script which relies on $SERVICE. So we need
- [ ] a new `run.sh`, service independent and which executes sh start.sh -?? (which option do you like? - on me
- [ ] a new `start.sh` (initially a different one for the two services) which the symlink part removed and the args - on you
- [ ] possibly an unified `start.sh` as a second step - on you

I'd keep this PR for the first item, since current start.sh will still work. So we can close it.
Sounds good ?

Reply:
sounds good! :) I will do my changes as part of https://github.com/dmwm/CRABServer/issues/6418 
Though, I did not get what you meant by:

> which option do you like?

Reply:
so we agree on `start.sh -r` for the "run the version in current". I am editing the table of actions above.